File: src/types/event.ts
// src/types/event.ts
export interface Location {
  lat: number;
  lng: number;
  address: string;
  place_id?: string; // Added place_id property
  city?: string;
  state?: string;
  postal_code?: string;
  country?: string;
}

export interface SportPosition {
  id: string;
  name: string;
  description?: string;
  quantity: number;
}

export interface Participant {
  user_id: string;
  user_name?: string;
  user_profile_image?: string;
  position_id?: string;
  position_name?: string;
  confirmed: boolean;
  joined_at?: string;
}

export interface Event {
  id: string;
  title: string;
  description: string;
  sport_type: string;
  skill_level: string; // beginner, intermediate, advanced
  start_time: string;
  end_time: string;
  location: Location;
  max_participants: number;
  participants: Participant[];
  positions?: SportPosition[];
  organizer_id: string;
  organizer_name: string;
  space_id?: string;
  space_name?: string;
  price_per_person?: number;
  is_private: boolean;
  created_at: string;
  updated_at: string;
}

File: backend/app/models/event.py
# backend/app/models/event.py
from typing import Optional, List, Any
from datetime import datetime
from pydantic import BaseModel, Field, EmailStr
from bson import ObjectId

from .user import PyObjectId, MongoBaseModel

# Modelos relacionados a eventos

class Location(BaseModel):
    lat: float
    lng: float
    address: str
    city: Optional[str] = None
    state: Optional[str] = None
    postal_code: Optional[str] = None

class SportPosition(BaseModel):
    name: str
    description: Optional[str] = None
    quantity: int = 1

class Participant(BaseModel):
    user_id: str
    user_name: Optional[str] = None
    user_email: EmailStr
    position_id: Optional[str] = None
    position_name: Optional[str] = None
    confirmed: bool = False
    joined_at: datetime = Field(default_factory=datetime.utcnow)

class Event(MongoBaseModel):
    title: str
    description: str
    sport_type: str
    skill_level: str
    start_time: datetime
    end_time: datetime
    location: Location
    max_participants: int
    participants: List[Participant] = []
    organizer_id: str
    organizer_name: Optional[str] = None
    space_id: Optional[str] = None
    space_name: Optional[str] = None
    price_per_person: float = 0
    is_private: bool = False
    positions: Optional[List[SportPosition]] = None
    created_at: datetime = Field(default_factory=datetime.utcnow)
    updated_at: datetime = Field(default_factory=datetime.utcnow)

    class Config:
        json_encoders = {
            ObjectId: str,
            datetime: lambda dt: dt.isoformat()
        }